test/parallel/test-no-extra-info-on-fatal-exception.js

```diff
@@ -0,0 +1,27 @@
+'use strict';
+// Flags: --no-extra-info-on-fatal-exception
+
+// This test ensures that the version is hidden at the end of stacktraces
+
+const common = require('../common');
+const assert = require('assert');
+const exec = require('child_process').exec;
+const fixtures = require('../common/fixtures');
+
+function errExec(script, callback) {
+  const cmd = `"${process.argv[0]}" --no-extra-info-on-fatal-exception "${fixtures.path(script)}"`;
+  return exec(cmd, (err, stdout, stderr) => {
+    // There was some error
+    assert.ok(err);
+
+    // More than one line of error output.
+    assert.ok(stderr.split('\n').length);
+
+    // Proxy the args for more tests.
+    callback(err, stdout, stderr);
+  });
+}
+
+errExec('throws_error.js', common.mustCall((err, stdout, stderr) => {
+  assert.match(stderr, /at node:internal\/main\/run_main_module:17:47\r?\n$/);
```

Comment:
Would it make more sense to add that test in `test/message` instead? This will break as soon as the `run_main_module` is updated.

Reply:
I think, it would make more sense to add this small single test in `test/parallel/test-error-reporting.js` file so we avoid the duplication of the method `errExec` as mentioned in https://github.com/nodejs/node/pull/38332#discussion_r650089711, what do you think?

Reply:
That would work too but it looks to me that `test/message` would still be the right place for this test:

> Tests in `test/message` pass if the output matches the expected output.
> Message tests are particularly useful when checking for exact error messages.

You just need to create `test/message/test-no-extra-info-on-fatal-exception.js` which contains:

```js
// Flags: --no-extra-info-on-fatal-exception
require('../common');

throw new Error('foo');
```

And then you need to create `test/message/test-no-extra-info-on-fatal-exception.out` which contains:

```
*:3
throw new Error('foo');
^

Error: foo
    at Object.<anonymous> (*:3:7)
    at Module._compile (node:internal/modules/cjs/loader:*:*)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:*:*)
    at Module.load (node:internal/modules/cjs/loader:*:*)
    at Function.Module._load (node:internal/modules/cjs/loader:*:*)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:*:*)
    at node:internal/main/run_main_module:*:*
```

If you prefer to put it in `test/parallel/test-error-reporting.js`, you need to make the test not dependent on the precise line number and column number for `node:internal/main/run_main_module` (using `:\d+:\d+` instead of `:17:47`).

Reply:
Indeed your first solution looks cleaner! :smile:
Done, updated!